### ISSUE TITLE
ComputeSecurityPolicy resource creation ignoring adaptiveProtectionConfig block

### ISSUE THREAD

**Issue by jfmainville** [GoogleCloudPlatform/k8s-config-connector#823]
### Checklist

- [X] I did not find a related open issue.
- [X] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [X] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Bug Description

While trying to deploy a new Cloud Armor security policy (`ComputeSecurityPolicy`) in our GCP infrastructure, I came across an issue related to the Adaptive Protection configuration (`adaptiveProtectionConfig`) block. When I use the following YAML configuration file to deploy the necessary security policy, the `adaptiveProtectionConfig` object isn't being applied and the Adaptive Protection setting stays to the Disabled state in the GCP console and the rest of the configuration is applied properly: 
```
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeSecurityPolicy
metadata:
  annotations:
    cnrm.cloud.google.com/project-id: <PROJECT_ID>
  name: security-policy
spec:
  adaptiveProtectionConfig:
    layer7DdosDefenseConfig:
      enable: true
      ruleVisibility: STANDARD
  description: "Cloud Armor security policy for the <SERVICE> service (managed by: config connector)"
  type: CLOUD_ARMOR
  rule:
    - action: allow
      priority: 2147483647
      match:
        versionedExpr: SRC_IPS_V1
        config:
          srcIpRanges:
            - "*"
```
Also, I checked the latest versions of the Config Connector as we have an older version (1.89.0) and there was no specific changes to the adaptiveProtectionConfig block since this release except for the addition of the `spec.adaptiveProtectionConfig.autoDeployConfig` field which shouldn't impact the implementation of the Adaptive Protection functionality to my knowledge.

### Additional Diagnostic Information

Here's the output that I get for the deployment of the configuration using the `kubectl describe computesecuritypolicy security-policy` command: 
```
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeSecurityPolicy
metadata:
  annotations:
    cnrm.cloud.google.com/management-conflict-prevention-policy: none
    cnrm.cloud.google.com/project-id: <PROJECT_ID>
    cnrm.cloud.google.com/state-into-spec: merge
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"compute.cnrm.cloud.google.com/v1beta1","kind":"ComputeSecurityPolicy","metadata":{"annotations":{"cnrm.cloud.google.com/project-id":"<PROJECT_ID>","cnrm.cloud.google.com/state-into-spec":"merge"},"labels":{"app":"<APP_NAME>","env":"<ENV_NAME>","tier":"<TIER_NAME>"},"name":"security-policy","namespace":"default"},"spec":{"adaptiveProtectionConfig":{"layer7DdosDefenseConfig":{"enable":true,"ruleVisibility":"STANDARD"}},"description":"Cloud Armor security policy for the <SERVICE_NAME> service (managed by: config connector)","rule":[{"action":"allow","match":{"config":{"srcIpRanges":["*"]},"versionedExpr":"SRC_IPS_V1"},"priority":2147483647}],"type":"CLOUD_ARMOR"}}
  creationTimestamp: "2023-06-13T14:28:04Z"
  finalizers:
  - cnrm.cloud.google.com/finalizer
  - cnrm.cloud.google.com/deletion-defender
  generation: 8
  labels:
    app: <APP_NAME>
    env: <ENV_NAME>
    tier: <TIER_NAME>
  name: security-policy
  namespace: default
  resourceVersion: "184879910"
  uid: d65e9c33-f8c8-4763-9977-e982f4ffd1b3
spec:
  description: 'Cloud Armor security policy for the <SERVICE> service (managed by:
    config connector)'
  resourceID: security-policy
  rule:
  - action: allow
    match:
      config:
        srcIpRanges:
        - '*'
      versionedExpr: SRC_IPS_V1
    priority: 2147483647
  type: CLOUD_ARMOR
status:
  conditions:
  - lastTransitionTime: "2023-06-13T14:41:10Z"
    message: The resource is up to date
    reason: UpToDate
    status: "True"
    type: Ready
  fingerprint: yI_1hnJ4Uhr=
  observedGeneration: 8
  selfLink: https://www.googleapis.com/compute/v1/projects/<PROJECT_ID>/global/securityPolicies/security-policy
```
As shown in the console output above, the deployment should include the Adaptive Protection Configuration block, but it's missing from the current configuration.

### Kubernetes Cluster Version

v1.24.12-gke.500

### Config Connector Version

1.89.0

### Config Connector Mode

cluster mode

### Steps to reproduce the issue

To reproduce the issue execute the following commands: 
1. `kubectl apply -f kubernetes/security-policy.yaml`
2. `kubectl describe computesecuritypolicy security-policy.yaml`

### YAML snippets

```YAML
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeSecurityPolicy
metadata:
  annotations:
    cnrm.cloud.google.com/project-id: <PROJECT_ID>
  name: security-policy
spec:
  adaptiveProtectionConfig:
    layer7DdosDefenseConfig:
      enable: true
      ruleVisibility: STANDARD
  description: "Cloud Armor security policy for the <SERVICE> service (managed by: config connector)"
  type: CLOUD_ARMOR
  rule:
    - action: allow
      priority: 2147483647
      match:
        versionedExpr: SRC_IPS_V1
        config:
          srcIpRanges:
            - "*"
```

**Comment by diviner524** [comment]
@jfmainville This was fixed in v1.90.0.

https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/v1.90.0

It seems you are installing KCC through GKE Add-on and as a result stuck at v1.89.0. If that's the case please consider switching to other installation methods. We have a comparison here:

https://cloud.google.com/config-connector/docs/concepts/installation-types#installation_methods

**Comment by jfmainville** [comment]
Hello, I just saw that in the releases, I wasn't using the correct attribute to search across the releases, sorry about that. This issue can be closed. Thanks for your help!
